src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -494,92 +481,53 @@ def _special_cases(self, text, cmd, outside):
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    else:
-                        for res_counter, _ in enumerate(results[0]):
-                            base = cmd_base
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    continue_flag = True
-
+            if len(args) == 1:  # if arguments start with query_symbol, just print query result
+                if args[0] == query_symbol:
+                    print(json.dumps(self.last.result, sort_keys=True, indent=2), file=self.output)
+                elif len(args) == 1 and args[0].startswith(query_symbol):
+                    query = args[0][symbol_len:]
+                    print(json.dumps(jmespath.search(query, input_dict), sort_keys=True, indent=2), file=self.output)
+            else:  # query, inject into cmd
+                def sub_result(arg):
+                    escaped_symbol = re.escape(query_symbol)
+
+                    def jmespath_query(match):
+                        query_result = jmespath.search(match.group(0)[symbol_len:], input_dict)
+                        if isinstance(query_result, list):
+                            return ' '.join(query_result)
+                        elif isinstance(query_result, (six.text_type, six.string_types)):
+                            return query_result
+                        else:  # query result is not a string or list, return invalid query message
+                            raise CLIError
+                    return re.sub(r'%s.*' % escaped_symbol, jmespath_query, arg)
+                cmd_base = ' '.join(map(sub_result, args))
+                self.cli_execute(cmd_base)
+            continue_flag = True
         except (jmespath.exceptions.ParseError, CLIError):
-            print("Invalid Query", file=self.output)
+            print("Invalid Query Input", file=self.output)
```

Comment:
Can the error message include which query was invalid?

Reply:
done